### DISCUSSION TITLE
blog/slice-insert/

### DISCUSSION THREAD

**Discussion by giscus** [zrcoder/leetcode#54] [General]
# blog/slice-insert/

假设有如下切片：
s := []int{2, 8 , 6, 9} 怎么在索引2处插入新元素100？变成：
{2, 8, 100, 6, 9} 朴素实现 // 1.s扩容，长度加一 s = append(s, 0) // 追加的元素值是什么不重要，只是为了扩容 // 2. 从待插入索引2开始，元素一一后移 for i := len(s)-1; i > 2; i-- { // 为了防止覆盖，从后向前遍历，每一位的值等于前一位的值 s[i] = s[i-1] } // 3.将要插入的100赋给s[2] s[2] = 100 一般化，不妨把这个逻辑提取成一个函数
// 在切片s的索引i处插入新元素v func insert(s *[]int, i, v int) { *s = append(*s, 0) for j := len(*s) - 1; j > i; j-- { (*s)[j] = (*s)[j-1] } (*s)[i] = v } 注意到函数接收的是切片的指针，有兴趣的话考虑下去掉*会怎么样？为什么会这样？

https://rdcode.netlify.app/blog/slice-insert/

**Comment by zrcoder** [comment]
go 在  1.21 版本后引入了 slices 包，里边有关于插入和删除的函数了～
